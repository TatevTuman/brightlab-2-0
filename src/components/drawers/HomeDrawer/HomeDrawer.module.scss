.home-drawer {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 1rem 0.5rem;
}
