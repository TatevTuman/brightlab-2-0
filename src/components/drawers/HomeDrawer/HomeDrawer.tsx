import React from 'react'
import { Drawer } from '@components'
import { useDrawer, useSiteMetadata } from '@hooks'
import { Link } from '@elements'
import styles from './HomeDrawer.module.scss'

export interface HomeDrawerProps {}

const HomeDrawer: React.FC<HomeDrawerProps> = props => {
  const drawerName = 'HomeDrawer'
  const { drawer, closeDrawer } = useDrawer(drawerName)
  const { navigation } = useSiteMetadata()

  return (
    <Drawer drawerName={drawerName}>
      <div className={styles.homeDrawer}>
        <nav role={'navigation'} data-direction="horizontal">
          <ul>
            {navigation.map(item => {
              const { path, label } = item

              return (
                <li key={path}>
                  <Link to={path}>{label}</Link>
                </li>
              )
            })}
          </ul>
        </nav>
      </div>
    </Drawer>
  )
}

export default HomeDrawer
