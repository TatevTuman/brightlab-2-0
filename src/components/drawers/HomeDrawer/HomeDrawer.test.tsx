import React, { useEffect } from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import { useModal } from '@hooks'
import HomeDrawer from './HomeDrawer'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

const TestHomeDrawer = () => {
  const { openModal } = useModal('HomeDrawer')

  useEffect(() => {
    openModal()
  }, [])

  return (
    <>
      <div id={'drawers'} />
      <HomeDrawer />
    </>
  )
}

describe('HomeDrawer', () => {
  it('renders correctly', async () => {
    const { container } = render(<TestHomeDrawer />)
    const awaitedContainer = await waitFor(() => container)

    expect(awaitedContainer).toMatchSnapshot()
  })
})
