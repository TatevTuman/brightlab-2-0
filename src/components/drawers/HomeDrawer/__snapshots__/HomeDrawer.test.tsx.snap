// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDrawer renders correctly 1`] = `
<div>
  <div
    id="drawers"
  />
</div>
`;
